src/table.rs

```diff
@@ -61,40 +62,351 @@ impl Table {
         self.river = Option::from(*deck_iterator.next().unwrap());
+    }
 
+    /// Mark all players that died from the last round as dead now
+    fn check_for_player_death(&mut self) {
+        // Check if players died on the past round
         for player in &mut self.players {
-            let card1 = *deck_iterator.next().unwrap();
-            let card2 = *deck_iterator.next().unwrap();
-            player.deal([card1, card2])
+            if player.death_hand_number.is_none() && player.total_money < self.ante {
+                player.death_hand_number = Some(self.hand_number)
+            }
         }
     }
 
+    /// Deal cards to the alive players and collect the ante from them.
+    fn deal_player_cards_collect_ante(&mut self, deck_iterator: &mut Iter<Card>) {
+        // Deal every alive player cards now
+        for player in &mut self.players {
+            if player.is_alive() {
+                let card1 = *deck_iterator.next().unwrap();
+                let card2 = *deck_iterator.next().unwrap();
+                player.deal([card1, card2]);
+                self.pot += player.bet(self.ante);
+                // the ante doesn't count as a turn so clarify the bot hasn't had a turn
+                player.has_had_turn_this_round = false;
+            }
+        }
+    }
+
+    pub fn get_state_string_for_player(&self, id: i8) -> JsonValue {
+        let player_strings: Vec<_> = self.players.iter().map(|x| if x.get_id() == id { x.as_json() } else { x.as_json_no_secret_data() }).collect();
+        object! {
+            flop: self.get_flop_string_secret(),
+            turn: self.get_turn_string_secret(),
+            river: self.get_river_string_secret(),
+            dealer_button_index: self.dealer_button_index,
+            players: player_strings,
+            hand_number: self.hand_number,
+        }
+    }
+    fn is_betting_over(&self) -> bool {
+        let all_players_bet_or_folded = self.check_all_players_ready_for_next_round();
+        let all_players_equal_bet = self.check_all_active_players_same_bet();
+        all_players_bet_or_folded && all_players_equal_bet
+    }
+    fn get_max_bet(&self) -> i32 {
+        self.players.iter().map(|x| match x.player_state {
+            PlayerState::Folded => { 0 }
+            PlayerState::Active(active_state) => { active_state.current_bet }
+        }).max().unwrap()
+    }
+    fn check_all_players_ready_for_next_round(&self) -> bool {
+        self.players.iter().map(|x| match x.player_state {
+            PlayerState::Folded => { true }
+            PlayerState::Active(_) => { x.has_had_turn_this_round }
+        }).reduce(|x, y| x && y).unwrap()
+    }
+    fn check_all_active_players_same_bet(&self) -> bool {
+        let max_bet = self.get_max_bet();
+        self.players.iter().map(|x| match x.player_state {
+            PlayerState::Folded => { true }
+            PlayerState::Active(a) => {
+                x.total_money < max_bet || a.current_bet == max_bet
```

Comment:
Should this be x.total_money == 0 instead of x.total_money < max_bet?

Reply:
This is right and has been fixed now.